grammar/Expr.g4

```diff
@@ -17,9 +17,20 @@ grammar Expr;
 /* To match the whole input */
 fullexpr: expr EOF;
 
-shift: TIME (op=('+' | '-') expr)?;
+shift: TIME shift_expr?;
+
+shift_expr
+    : op=('+' | '-') NUMBER                    # signedNumber
+    | shift_expr op=('+' | '-') expr           # shiftAddsub
+    | op=('+' | '-') IDENTIFIER                # signedIdentifier
+    | op=('+' | '-') IDENTIFIER '.' IDENTIFIER # signedPortField
```

Comment:
This is now "partially" forbidden.
I left a TODO comment in the grammar because there is some space for improvements on that matter, but the main bug is solved.